MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflCommand.h

```diff
@@ -33,12 +35,27 @@ Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
 class ReflCommand {
 public:
+  ReflCommand() : m_child(){};
   virtual ~ReflCommand(){};
 
   virtual void execute() = 0;
   virtual std::string name() = 0;
   virtual std::string icon() = 0;
+  virtual bool hasChild() final { return !m_child.empty(); };
+  virtual void setChild(std::vector<std::unique_ptr<ReflCommand>> child) final {
+    m_child = std::move(child);
+  }
+  virtual const std::vector<std::unique_ptr<ReflCommand>> &getChild() final {
+    return m_child;
+  }
+	virtual bool isSeparator() final { return name().empty() && icon().empty(); }
```

Comment:
My intention here was precisely to dictate derived classes how to implement the `isSeparator()` method: I thought that a `ReflCommand` with no name and no icon should be a separator. Is this a bad idea? Would it be better that each derived class implements its own `isSeparator()`, if so why?